File: AlwaysPrintProject/Cloud/frontend/src/types/log-analysis.ts
/**
 * Tipos relacionados con el análisis de logs de workstations.
 */

export interface LogAnalysisResponse {
  id: string
  workstation_id: string
  organization_id: string
  analysis_date: string
  analysis_text: string
  processing_path: 'direct' | 'structural'
  log_size_bytes: number
  processing_duration_ms: number
  original_filename: string
  created_at: string
  updated_at: string
}

export interface LogAnalysisTodayCheckResponse {
  exists: boolean
  analysis_id: string | null
  analysis_date: string | null
}

export interface LogAnalysisListResponse {
  items: LogAnalysisResponse[]
  total: number
  page: number
  page_size: number
}

File: AlwaysPrintProject/Cloud/backend/app/schemas/log_analysis.py
"""
Schemas Pydantic para análisis de logs de workstations.

Define los schemas de respuesta para:
- LogAnalysisResponse: resultado completo de un análisis
- LogAnalysisTodayCheckResponse: verificación de análisis existente del día
- LogAnalysisListResponse: lista paginada de análisis
"""

from datetime import date, datetime
from typing import Optional
from uuid import UUID

from pydantic import BaseModel, Field


# === SCHEMAS DE RESPONSE ===


class LogAnalysisResponse(BaseModel):
    """Schema de respuesta para un análisis de log individual."""

    id: UUID
    workstation_id: UUID
    organization_id: UUID
    analysis_date: date = Field(..., description="Fecha del análisis (día del log)")
    analysis_text: str = Field(..., description="Texto de respuesta del LLM con el análisis")
    processing_path: str = Field(
        ..., description="Ruta de procesamiento utilizada: 'direct' o 'structural'"
    )
    log_size_bytes: int = Field(..., description="Tamaño del log procesado en bytes")
    processing_duration_ms: int = Field(
        ..., description="Duración total del procesamiento en milisegundos"
    )
    original_filename: str = Field(..., description="Nombre original del archivo de log")
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class LogAnalysisTodayCheckResponse(BaseModel):
    """Schema de respuesta para verificar si existe un análisis del día actual."""

    exists: bool = Field(..., description="Si existe un análisis para hoy")
    analysis_id: Optional[UUID] = Field(
        None, description="ID del análisis existente, None si no existe"
    )


class LogAnalysisListResponse(BaseModel):
    """Schema de respuesta para lista paginada de análisis de una workstation."""

    items: list[LogAnalysisResponse]
    total: int = Field(..., description="Total de análisis disponibles")
    page: int = Field(..., description="Página actual (1-based)")
    page_size: int = Field(..., description="Tamaño de página")
